Feature/auth

escomputer opened the pull request:
## 📌 관련 이슈
- Closes #536 <!-- 연결된 이슈 번호 기재 -->



## ✨ 기능 요약

| 항목 | 내용 |
|------|------|
| 🆕 기능명 | 회원가입,로그인,로그아웃,소셜로그인 |
| 🔍 목적 | 경매 입찰자 및 관리자들의 인증/인가를 위하여|
| 🛠️ 변경사항 | 구글,네이버 소셜로그인 , 로컬과 구분됨 자세한건 커밋메시지 참고|



## 📝 상세 내역

| 번호 | 내용 |
|------|------|
| 1️⃣ | 주요 로직/컴포넌트/서비스 등 변경 설명 |
| 2️⃣ | 관련 유틸, 공통 로직 수정 여부 |
| 3️⃣ | 리팩토링 또는 제거된 불필요 코드 |

---

✅ 테스트 체크리스트
 테스트 할 내용이 있다면 작성해주세요
- [ ] 소셜로그인 시에 브라우저에 접속후 JSESSIONID 복사해서 postman cookie 헤더에 넣어주신 뒤 테스트 바랍니다.
- [ ] 에러사항 체크 부탁드립니다 ( 인증번호 5분 내에 재전송안됨, 로그인 5회 실패시 5분 기다려야됨)

---

## 📸 포스트맨 캡처 사진

![image](https://github.com/user-attachments/assets/58cc6a4f-c584-4485-96f8-a33620003a1e)

---

## 🙋‍♀️ 리뷰어 참고사항

- 테스트 방법 -

구글 -http://localhost:8080/oauth2/authorization/google
(구글은 테스트 시에는, 테스트 가능한 계정을 등록해야합니다. 
팀원분들 구글이메일 알려주시면 등록 후 테스트 가능하게 해드리겠습니다!)

네이버-http://localhost:8080/oauth2/authorization/naver

로그인 후 에 관리자모드 -application - jessionid 복사후

포스트맨에 헤더에 넣어야 로컬처럼 로그인상태/로그아웃상태 판별가능합니다.

escomputer committed:
move : user 산하에 auth 패키지로 구조 변경

escomputer committed:
init : Oauth2 gradle 설정

escomputer committed:
refactor : User에 provider(소셜) 추가, 지원하지않는 소셜로그인 에러코드추가

escomputer committed:
feat : google,naver,kakao 사용자 정보 파싱 클래스 및 파싱 인터페이스

escomputer committed:
feat : 소셜 로그인 처리 서비스

p.s. 병합 후 imageservice 에러 해결

escomputer committed:
feat : 소셜 로그인 사용자 (정보)

escomputer committed:
feat : securityconfig에 소셜로그인설정, 성공시 프론트 전달 파라미터

escomputer committed:
refactor : 오탈자 + security를 위하여 ROLE_ 로 변경

escomputer committed:
INIT : 소셜로그인 PROPERTIES 추가

escomputer committed:
remove : 카카오랑 jwt날리기

escomputer committed:
feat : 소셜 로그인 (naver,google)/
회원가입 (인증번호 전송 (5분 동안은 재전송안됨), 검증)/
로그인(로그아웃상태여야만 가능함,일정횟수(지금 5회) 이상 틀릴시에, 5분 뒤에 가능),
로그아웃( 로그인상태가 아닌데 요청할시 에러발생)

5분 기다리는거는 ttl사용해서 몇분 몇초남았는지 띄워집니다.

escomputer committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/auth

# Conflicts:
#	src/main/java/com/example/auction/common/exception/ErrorCode.java
#	src/main/java/com/example/auction/config/SecurityConfig.java
#	src/main/java/com/example/auction/filter/JwtAuthenticationFilter.java
#	src/main/resources/application.properties

escomputer committed:
refactor : 오탈자 수정

nagul2 commented on src/main/java/com/example/auction/common/response/CommonResponse.java:
```diff
@@ -87,4 +88,26 @@ public static CommonResponse<Void> error(BaseCode baseCode, List<ErrorResponse.F
 			.build();
 	}
 
+	// 커스텀 메시지가 있을때
+	public static <T> CommonResponse<T> error(BaseCode code, String customMessage) {
+		return CommonResponse.<T>builder()
+			.isError(true)
+			.status(code.getHttpStatus())
+			.code(code.getCode())
+			.message(customMessage) // 커스텀 메시지 우선
+			.build();
+	}
+
+	public static <T> CommonResponse<T> success(SuccessCode successCode) {
+		return CommonResponse.<T>builder()
+			.timestamp(LocalDateTime.now())
+			.isError(false)
+			.status(successCode.getHttpStatus())
+			.code(successCode.getCode())
+			.message(successCode.getMessage())
+			.data(null)
+			.errorResponse(null)
+			.build();
+	}
+
```
이부분은 이미 만들어두신 error코드나 CommonResponse.ok(), create()로 충분이 구현이 되었을 것 같은데 추가로 생성하신 이유가 있을까요??

이미 기존 코드에 SuccessCode나, BaseCode를 활용하고 있기에 같은 기능을하는 다양한 공통 객체 생성이 이루어지는 것 같아 궁금합니당

escomputer replied:
ok는 basecode 를 매개변수로 받는 방식의 메서드가 없는데, 로그아웃이 스프링 시큐리티에서 관리하는거라 핸들러가 필요하더라고요 ! 거기서 로그아웃 메시지 처리를 하려면 공통응답+ 코드 가 필요해서 하려고했는데 그냥 200 ok만 던져도 되게끔 수정할까요??

nagul2 replied:
설명 듣고 이해했습니다!
커스텀이 필요하시다면 추가하는게 맞을 것 같습니다.

mo00ai commented on src/main/resources/application.properties:
```diff
@@ -15,6 +15,38 @@ server.error.include-message=always
+#session
+spring.session.store-type:redis
+spring.session.timeout=30m
+# multipartfile
+file.upload-dir=C:\projects
```
이 부분 
file.upload-dir=${IMAGE_DIR}

이렇게 바꾸고 각자 환경변수에서 값 주입하는 게 좋을 것 같아요!
로컬에서 파일 저장하는 위치가 다들 다르기 때문에 이렇게 한다고 하더라구요!!

escomputer replied:
앗 네 ! 
제거 테스트 해보느라, 제 프로젝트 파일 경로로 설정해놨다가 고치는 걸 깜빡했네요ㅠㅠ 감사합니다 !

nagul2 commented on src/main/java/com/example/auction/config/security/SecurityConfig.java:
```diff
@@ -0,0 +1,87 @@
+				.accessDeniedHandler((request, response, accessDeniedException) -> {
+					response.sendError(403, "Access Denied: " + accessDeniedException.getMessage());
+				})
+			);
```
이거 sendError로 하면 json 응답으로 안갈건데 확인한번 해보시는 것이 좋을 것 같습니다.
기존에 JwtAuthenticationFilter에 적용한 예외 응답 방식으로 처리해야 필터의 예외는 정상적으로 json 응답이 되는걸로 알고있는데 한번 확인 부탁드릴게요!

escomputer replied:
jwtAuthenticationFilter보면서 그 방식으로 변경했는데 확인 한번만 부탁드리겠습니다!!!

nagul2 replied:
잘 된것 같습니다~

escomputer committed:
refactor : 로컬 파일 위치에서 환경변수로 설정

nagul2 commented on src/main/java/com/example/auction/domain/user/auth/controller/AuthController.java:
```diff
@@ -0,0 +1,127 @@
+		return CommonResponse.ok();
+	}
+
+	@GetMapping("/users/me")
+	public CommonResponse<?> me(@AuthenticationPrincipal AuthUserWrapper wrapper) {
+		if (wrapper == null) {
+			return CommonResponse.error(AuthErrorCode.NOT_AUTHENTICATED);
+		}
+		return CommonResponse.ok("로그인된 유저 ID: " + wrapper.getUser().getId());
+	}
+
+}
```
이건 테스트 용도일까요?
만약 그렇다면 주석으로 구분해주시면 보기 좋을 것 같습니다!

escomputer replied:
구분했습니다!

nagul2 commented on src/main/java/com/example/auction/domain/user/auth/service/EmailService.java:
```diff
@@ -0,0 +1,110 @@
+public class EmailService {
+
+	private final JavaMailSender mailSender;
+	private final RedisTemplate<String, String> redisTemplate;
+
```
나중에 RedisService에 있는 메서드들을활용하도록 변경하는게 좋을 것 같아요~

escomputer replied:
넵 한번 해보겠습니다

escomputer committed:
docs : 테스트용 메서드 구분 주석 추가

escomputer committed:
fix : 코드 형식의 응답에서, json 응답으로 변환하기 위한 메서드 분리 및 objectMapper사용

escomputer committed:
refactor : 로그아웃 공통응답으로 수정

escomputer committed:
refactor : 로그아웃 공통응답으로 수정

escomputer committed:
refactor : 세션변경으로 인한 에러코드 수정 및, 공통응답객체 통일, 코드 컨벤션 적용으로 import문 순서 조정 됨

escomputer committed:
refactor : redisTemplate -> RedisService의 메서드,
인증번호, 로그인 등 이메일 관련해서 몇분 몇초 남았다는 예외를 던질때 커스텀 메시지가 필요해서 customexception 및 공통응답객체 리팩토링
